siliconcompiler/schema/schema_cfg.py

```diff
@@ -763,42 +764,465 @@ def schema_pdk(cfg, stackup='default'):
+
+    scparam(cfg, ['datasheet', 'proc', name, 'features'],
+            sctype='[str]',
+            shorthelp="Datasheet: processor features",
+            switch="-proc_features 'name <str>'",
+            example=[
+                "cli: -proc_features '0 SIMD'",
+                "api: chip.set('datasheet','proc','cpu','features', 'SIMD')"],
+            schelp="""List of maker specified processor features.""")
+
+    scparam(cfg, ['datasheet', 'proc', name, 'datatypes'],
+            sctype='[enum]',
+            enum=['int1', 'int4', 'int8', 'int16', 'int32', 'int64', 'int128',
+                  'sint4', 'sint8', 'sint16', 'sint32', 'sint64', 'sint128',
+                  'bfloat16', 'fp16', 'fp32', 'fp64'],
+            shorthelp="Datasheet: processor datatypes",
+            switch="-proc_datatypes 'name <str>'",
+            example=[
+                "cli: -proc_datatypes '0 SIMD'",
+                "api: chip.set('datasheet','proc','cpu','features', 'SIMD')"],
+            schelp="""List of maker specified processor features.""")
+
+    metrics = {'archsize': ['architecture size', 64, None],
+               'cores': ['number of cores', 4, None],
+               'speed': ['maximum frequency', 100, 'MHz'],
+               'ops': ['operations per cycle per core', 4, None],
+               'mults': ['hard multiplier units per core', 100, None],
+               'icache': ['l1 icache size', 32, 'KB'],
+               'dcache': ['l1 dcache size ', 32, 'KB'],
+               'l2cache': ['l2 cache size', 1024, 'KB'],
+               'l3cache': ['l3 cache size', 1024, 'KB'],
+               'sram': ['shared sram', 128, 'KB'],
+               'nvm': ['shared non-volatile memory', 128, 'KB']}
+
+    for i, v in metrics.items():
+        scparam(cfg, ['datasheet', 'proc', name, i],
+                unit=v[2],
+                sctype='int',
+                shorthelp=f"Datasheet: processor {v[0]}",
+                switch=f"-proc_{i} 'name <int>'",
+                example=[
+                    f"cli: -proc_{i} 'name {v[1]}'",
+                    f"api: chip.set('datasheet', 'proc', name, '{i}', {v[1]})"],
+                schelp=f"""Processor {v[1]}.""")
+
+    ######################
+    # FPGA
+    ######################
+
+    scparam(cfg, ['datasheet', 'fpga', 'arch'],
+            sctype='str',
+            shorthelp="Datasheet: fpga architecture",
+            switch="-fpga_arch '<str>'",
+            example=[
+                "cli: -fpga_arch 'openfpga'",
+                "api: chip.set('datasheet', 'fpga', 'arch', 'openfpga')"],
+            schelp="""FPGA architecture.
+            """)
+
+    metrics = {'luts': ['LUTs (4-input)', 32000, None],
+               'registers': ['registers', 100, None],
+               'plls': ['pll blocks', 1, None],
+               'mults': ['hard multiplier units', 100, None],
+               'totalram': ['totaldistributed ram', 128, 'Kb'],
+               'distram': ['distributed ram', 128, 'Kb'],
```

Comment:
how is total distributed different from distram?

Reply:
fixed, total=dist+block